utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -797,6 +775,59 @@ impl<'trie, T: TrieValue + Into<u32>> CodePointTrie<'trie, T> {
             cpm_range: init_range,
         }
     }
+
+    /// Returns a [`UnicodeSet`] for the code points that have the given
+    /// [`TrieValue`] in the trie.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_codepointtrie::planes;
+    ///
+    /// let trie = planes::get_planes_trie();
+    ///
+    /// let plane_val = 2;
+    /// let sip = trie.get_set_for_value(plane_val as u8);
+    ///
+    /// let start = plane_val * 0x1_0000;
+    /// let end = start + 0xffff;
+    ///
+    /// assert!(!sip.contains_u32(start - 1));
+    /// assert!(sip.contains_u32(start));
+    /// assert!(sip.contains_u32(end));
+    /// assert!(!sip.contains_u32(end + 1));
+    /// ```
+    pub fn get_set_for_value(&self, value: T) -> UnicodeSet<'static> {
+        let value_ranges = self
+            .iter_ranges()
+            .filter(move |cpm_range| cpm_range.value == value)
+            .map(|cpm_range| RangeInclusive::new(*cpm_range.range.start(), *cpm_range.range.end()));
```

Comment:
I think you can just do

```suggestion
            .map(|cpm_range| cpm_range.range);
```